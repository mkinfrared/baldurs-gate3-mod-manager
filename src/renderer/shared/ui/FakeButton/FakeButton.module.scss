@use "../../styles/_vars.scss" as v;

.FakeButton {
  height: 100%;
  width: 100%;

  &:focus-visible {
    @include v.focus-visible;
  }
}
