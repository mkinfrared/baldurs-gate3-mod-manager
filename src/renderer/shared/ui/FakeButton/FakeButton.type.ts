import { MouseEvent, ReactNode } from "react";

export type FakeButtonProps = {
  ariaLabel?: string;
  ariaLabelledby?: string;
  autoFocus?: boolean;
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  children?: ReactNode;
  onClick?: (event: MouseEvent) => void;
};
