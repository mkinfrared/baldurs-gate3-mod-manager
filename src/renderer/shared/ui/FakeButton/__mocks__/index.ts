export { default } from "./FakeButton.mock";
