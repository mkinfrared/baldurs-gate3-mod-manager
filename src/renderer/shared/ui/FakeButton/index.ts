export { default } from "./FakeButton";

export * from "./FakeButton.type";
