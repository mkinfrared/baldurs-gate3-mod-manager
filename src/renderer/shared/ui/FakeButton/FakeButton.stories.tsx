import { Meta, StoryObj } from "@storybook/react";

import FakeButton from "./FakeButton";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/FakeButton",
  component: FakeButton,
  tags: ["autodocs"],
} satisfies Meta<typeof FakeButton>;

const Default: Story = {
  render: (args) => (
    <div className={css.Story}>
      <FakeButton {...args} />
    </div>
  ),
};

export { Default };

export default meta;
