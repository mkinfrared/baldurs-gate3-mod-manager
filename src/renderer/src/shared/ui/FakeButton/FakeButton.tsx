import { KeyboardEventHandler, useEffect, useRef } from "react";

import { classNames } from "@renderer/shared/lib/helpers";

import css from "./FakeButton.module.scss";
import { FakeButtonProps } from "./FakeButton.type";

const FakeButton = ({
  ariaLabel,
  ariaLabelledby,
  autoFocus,
  className,
  children,
  onClick,
}: FakeButtonProps) => {
  const ref = useRef<HTMLDivElement>(null);

  const handleKeyDown: KeyboardEventHandler = (event) => {
    const { key } = event;

    if (key === "Enter" || key === " ") {
      const { target } = event;
      const ev = new MouseEvent("click", { bubbles: true });

      target.dispatchEvent(ev);
    }
  };

  useEffect(() => {
    if (autoFocus && ref.current) {
      ref.current.autofocus = true;
    }
  }, []);

  return (
    <div
      ref={ref}
      className={classNames(css.FakeButton, className)}
      onClick={onClick}
      onKeyDown={handleKeyDown}
      role="button"
      tabIndex={0}
      data-testid="FakeButton"
      aria-label={ariaLabel}
      aria-labelledby={ariaLabelledby}
    >
      {children}
    </div>
  );
};

export default FakeButton;
