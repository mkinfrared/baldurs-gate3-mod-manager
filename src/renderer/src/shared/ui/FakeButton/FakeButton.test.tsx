import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import FakeButton from "./FakeButton";

describe("<FakeButton />", () => {
  const Component = <FakeButton />;

  it("should be defined", () => {
    expect(FakeButton).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("FakeButton");

    expect(element).toBeDefined();
  });
});
