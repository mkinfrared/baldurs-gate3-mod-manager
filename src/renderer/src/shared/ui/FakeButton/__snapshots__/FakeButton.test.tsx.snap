// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<FakeButton /> > should match the snapshot 1`] = `
<div>
  <div
    class="FakeButton"
    data-testid="FakeButton"
    role="button"
    tabindex="0"
  />
</div>
`;
